workshop.wwwapi/Endpoints/SurgeryEndpoint.cs

```diff
@@ -11,23 +12,108 @@ public static void ConfigurePatientEndpoint(this WebApplication app)
             var surgeryGroup = app.MapGroup("surgery");
 
             surgeryGroup.MapGet("/patients", GetPatients);
+            surgeryGroup.MapGet("/patientById/{id}", GetPatientById);
             surgeryGroup.MapGet("/doctors", GetDoctors);
+            surgeryGroup.MapGet("/doctorsById/{id}", GetDoctorsById);
             surgeryGroup.MapGet("/appointmentsbydoctor/{id}", GetAppointmentsByDoctor);
+            surgeryGroup.MapGet("/appointmentsbypatient/{id}", GetAppointmentsByPatient);
+            surgeryGroup.MapGet("/appointmentsbycompositeid/{doctorid} {patientid} {booking}", GetAppointmentsByCompositeId);
+            surgeryGroup.MapPost("/createPatient", CreatePatient);
+            surgeryGroup.MapPost("/createDoctor", CreateDoctor);
+            surgeryGroup.MapPost("/createAppointment", CreateAppointment);
+
+        }
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public static async Task<IResult> CreatePatient(IRepository repository, string FullName)
+        {
+            Patient? patient = await repository.CreatePatient(FullName);
+            if(patient == null){
+                return TypedResults.NotFound();
```

Comment:
why not found? it should be BadRequest because you failed to create; you are not searching for a patient because youre creating one